test/distributed/fsdp/test_fsdp_dtensor_state_dict.py

```diff
@@ -126,12 +150,17 @@ def test_dtensor_sharded_tensor_state_dict_identical(self, offload_to_cpu):
             k2, v2 = sharded_tensor_sd
             self.assertEqual(k1, k2)
 
-            self.assertEqual(type(v1), DTensor)
-            self.assertEqual(type(v2), ShardedTensor)
-            # check whether local_tensor are the same
-            self.assertEqual(v1.to_local(), v2.local_tensor())
-            # check whether device are the same
-            self.assertEqual(v1.to_local().device, v2.local_tensor().device)
+            # if the ShardedTensor is an empty shard,
+            # then the local tensor of DTensor should be local_tensor=tensor([])
+            if len(v2.local_shards()) == 0:
+                self.assertEqual(v1.to_local().numel(), 0)
```

Comment:
why we are comparing the DTensor path with sharded tensor path? Can be done later but I think we should probably compare with the full state dict path, otherwise the sharded tensor path might become stickier when we get rid of it?